File: cuantizacion.py
import numpy as np
import matplotlib.pyplot as plt

def cuantizacion(x,N,H):
    minimo = min(x)
    x = x - minimo
    for i in np.arange(0,len(x),1):
        #x[i] -= minimo
        if x[i] < 0:
            x[i] = 0
        elif (x[i] >= H*int(x[i]/H) and x[i] < (N-1)*H):
            x[i] = H*np.round(x[i]/H)
        else:
            x[i] = (N-1)*H
        #x[i] += minimo
    x = x + minimo
    return x

File: cuantizacion.m
function [y]=cuantizacion(x,N,H)
  #N numero de niveles
  #H magnitud del paso
  #a la señal le sumamos un valor arbitrario para aplicar cuantizacion
  minimo = min(x);
  x = x - minimo; #resto minimo
  y= 0.*(x<0) + ...
     H.*fix(x./H).*(x >= 0 & x < (N-1).*H) + ...
     (N-1).*H .* (x >= (N-1).*H);
  y = y + minimo;#sumo minimo
  endfunction
